Framework/PythonInterface/plugins/algorithms/AlignComponents.py

```diff
@@ -499,7 +515,641 @@ def _extract_tofs(self, table_tofs: TableWorkspace) -> np.ndarray:
+        #             return component.getID()
+        #         else:
+        #             return self._getLastDetID(component[component.nelements() - 1])
+        #
+        #     def _mapOptions(self, inX):
+        #         """
+        #         Creates an array combining the refining and constant variables
+        #         This is required because scipy.optimise.minimize expects a constant
+        #         number of variables, so need to be able to maps any number of
+        #         inputs to six outputs.
+        #
+        #         @param inX :: list of length 3 or 6
+        #
+        #         @return list of length 6
+        #         """
+        #         x0_index = 0
+        #         out = []
+        #         for opt in self._optionsList:
+        #             if self._optionsDict[opt]:
+        #                 out.append(inX[x0_index])
+        #                 x0_index += 1
+        #             else:
+        #                 out.append(self._initialPos[self._optionsList.index(opt)])
+        #         return out
+        #
+        #     def _eulerToQuat(self, alpha, beta, gamma, convention):
+        #         """
+        #         Convert Euler angles to a quaternion
+        #         """
+        #         getV3D = {'X': V3D(1, 0, 0), 'Y': V3D(0, 1, 0), 'Z': V3D(0, 0, 1)}
+        #         return (Quat(alpha, getV3D[convention[0]]) * Quat(beta, getV3D[convention[1]])
+        #                 * Quat(gamma, getV3D[convention[2]]))
+        #
+        #     def _eulerToAngleAxis(self, alpha, beta, gamma, convention):
+        #         """
+        #         Find the Euler axis and Euler angle
+        #
+        #         @param alpha :: rotation angle, in degrees, around the first axis of `convention`
+        #         @param beta :: rotation angle, in degrees, around the second axis of `convention`
+        #         @param gamma :: rotation angle, in degrees, around the third axis of `convention`
+        #         @param convention :: string, e.g. 'YZX'
+        #
+        #         @return Euler angle, and three direct cosines defining the Euler axis
+        #         """
+        #         quat = self._eulerToQuat(alpha, beta, gamma, convention)
+        #         if quat[0] == 1:
+        #             return 0, 0, 0, 1
+        #         deg = math.acos(quat[0])
+        #         scale = math.sin(deg)
+        #         deg *= 360.0 / math.pi
+        #         ax0 = quat[1] / scale
+        #         ax1 = quat[2] / scale
+        #         ax2 = quat[3] / scale
+        #         return deg, ax0, ax1, ax2
+        #
+        #
+        # try:
+        #     from scipy.optimize import minimize, OptimizeResult
+        #     AlgorithmFactory.subscribe(AlignComponents)
+        # except ImportError:
+        #     logger.debug('Failed to subscribe algorithm AlignComponets; cannot import minimize from scipy.optimize')eak centers (d-spacing units)
```

Comment:
Is there any particular reason we want to keep the outdated code as comments?

Reply:
This is some legacy "code", by Ross, I believe. I guess we can part with it at this time

Reply:
Actually, it's me who did this...yesterday!
I have no clue how this happened.